app/assets/javascripts/components/gtm-form-listener.js

```diff
@@ -0,0 +1,35 @@
+function GTMFormListener ($form, $dataLayer) {
+  this.$form = $form
+  this.dataLayer = $dataLayer || window.dataLayer
+}
+
+GTMFormListener.prototype.handleSubmit = function (event) {
+  var eventName = this.$form.dataset.gtm
+  var message = {}
+
+  var inputElements = this.$form.querySelectorAll('input:checked')
+  inputElements.forEach(function (element) {
+    if (element.value.length > 0) {
+      message = {}
+      message[eventName] = {}
+      message[eventName][element.name] = element.value
```

Comment:
Is this code ok for a set of checkboxes with the same name too? e.g. "field[]=foo&field[]=bar"

Reply:
Yes and added a spec in 569a19254b15d163e49cc45c2b0ad2e0087c89ec